src/pkg/exporter/exporter.go

```diff
@@ -0,0 +1,97 @@
+package exporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/goharbor/harbor/src/lib/log"
+)
+
+// Opt is the config of Harbor exporter
+type Opt struct {
+	Port                   int
+	MetricsPath            string
+	ExporterMetricsEnabled bool
+	MaxRequests            int
+	TLSEnabled             bool
+	Certificate            string
+	Key                    string
+	CacheTime              int64
+	CacheCleanInterval     int64
+}
+
+// NewExporter creates a exporter for Harbor with the configuration
+func NewExporter(opt *Opt) *Exporter {
+	exporter := &Exporter{
+		Opt:        opt,
+		Collectors: make(map[string]prometheus.Collector),
+	}
+	if opt.CacheTime > 0 {
+		CacheInit(opt)
+	}
+	exporter.RegisterCollector(healthCollectorName, NewHealthCollect(hbrCli))
+	exporter.RegisterCollector(systemInfoCollectorName, NewSystemInfoCollector(hbrCli))
+	exporter.RegisterCollector(ProjectCollectorName, NewProjectCollector())
+	r := prometheus.NewRegistry()
+	r.MustRegister(exporter)
+	exporter.Server = newServer(opt, r)
+
+	return exporter
+}
+
+// Exporter is struct for Harbor which can used to connection Harbor and collecting data
+type Exporter struct {
+	*http.Server
+	Opt        *Opt
```

Comment:
Exposing Opt may be confusing, what if you intialized an Exporter and later the Opt of an Export is updated?  This will cause inconsistency.